sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java

```diff
@@ -1130,6 +1122,12 @@ public DeleteKey withProjectId(ValueProvider<String> projectId) {
 
     @Override
     public PDone expand(PCollection<T> input) {
+      checkArgument(projectId != null, "withProjectId() is required");
+      if (projectId.isAccessible()) {
```

Comment:
Unfortunately currently validate() is also called at construction time, and the runtime options are still not available.